Mage/src/main/java/mage/abilities/effects/common/cost/SpellsCostReductionControllerEffect.java

```diff
@@ -71,14 +82,20 @@ protected SpellsCostReductionControllerEffect(final SpellsCostReductionControlle
         this.amount = effect.amount;
         this.manaCostsToReduce = effect.manaCostsToReduce;
         this.upTo = effect.upTo;
+        this.convertToGeneric = effect.convertToGeneric;
+        this.staticText = effect.staticText;
     }
 
     @Override
     public boolean apply(Game game, Ability source, Ability abilityToModify) {
+        int reductionAmount = this.amount.calculate(game, source, this);
         if (manaCostsToReduce != null) {
-            CardUtil.adjustCost((SpellAbility) abilityToModify, manaCostsToReduce, false);
+            ManaCosts<ManaCost> calculatedManaCostsToReduce = new ManaCostsImpl<>();;
```

Comment:
There is `upTo` mode to choose any amount to reduce. Current code:
* colored reduce + upTo -- non implemented
* generic reduce + upTo -- implemented

I recommends to add runtime error in constructor like
``` java
        // runtime check
        if (this.manaCostsToReduce != null && this.upTo) {
            throw new IllegalStateException("Wrong code usage: colored cost reduce un-support "up to" mode");
        }
```

Reply:
No constructs set both of those parameters, so with the current code that can't happen.